### ISSUE TITLE
Invalid content-length in custom ErrorHandler

### ISSUE THREAD

**Issue by briceruzand** [fastify/fastify#2543]
## 🐛 Bug Report

I add a custom error handler to add some extra data to my api answer (stack, ...).
- I send a request too an elasticsearch server, which reply an answer (an error) with a content-length of 335 bytes.
- In my error handler, I answer with bigger body (more than 335 bytes content-length).

## To Reproduce

If a thrown error have headers.'content-length' properties it is propagated to the fastify answer like #2214 but with an errorHandler.

```js
import fastify from 'fastify';

const server = fastify();

server.get('/', async (request, reply) => {
  const error = new Error('MY_ERROR_MESSAGE');
  error.headers = {
    'content-length': 2,
  };
  throw error;
});

server.setErrorHandler((error, request, reply) => {
  reply.code(500).send({ error: 'longer than 2' });
});

const start = async () => {
  await server.listen(3000, '0.0.0.0');
};

start();
```

```
curl -v http://127.0.0.1:3000/
*   Trying 127.0.0.1...
* Connected to 127.0.0.1 (127.0.0.1) port 3000 (#0)
> GET / HTTP/1.1
> Host: 127.0.0.1:3000
> User-Agent: curl/7.47.0
> Accept: */*
> 
< HTTP/1.1 500 Internal Server Error
< content-length: 2
< content-type: application/json; charset=utf-8
< Date: Tue, 08 Sep 2020 13:33:51 GMT
< Connection: keep-alive
< Keep-Alive: timeout=5
< 
* Excess found in a non pipelined read: excess = 23, size = 2, maxdownload = 2, bytecount = 0
* Connection #0 to host 127.0.0.1 left intact

```

## Expected behavior

Answer generated by my error handler should have the right content length.
It has the following side effects : 
- if my answer is smaller than 2 bytes : it hangs for 5secs
- if my answer is bigger than 2 bytes : my answer is truncated and my json is not valid.

## Your Environment

- *node version*: 14
- *fastify version*: >=3.3.0
- *os*: Linux

Thx

**Comment by mcollina** [comment]
Would you like to send a Pull Request to address this issue? Remember to add unit tests.

**Comment by briceruzand** [comment]
I don't know how too reproduce it in a smaller env than the one I have provided.
May be an internal point of view you will find the cause.

**Comment by briceruzand** [comment]
It looks like #2214

**Comment by briceruzand** [comment]
After looking at #2214 : 
I found simpler way to reproduce it, like in #2214 with an errorHandler.

```js
import fastify from 'fastify';

const server = fastify();

server.get('/', async (request, reply) => {
  const error = new Error('MY_ERROR_MESSAGE') as any;
  error.headers = {
    'content-length': 2,
  };
  throw error;
});

server.setErrorHandler((error, request, reply) => {
  reply.code(500).send({ error: 'longer than 2' });
});

const start = async () => {
  await server.listen(3000, '0.0.0.0');
};

start();
```

**Comment by mcollina** [comment]
Would you like to send a PR?

**Comment by briceruzand** [comment]
I provide only test case for the bug not the fix  in PR #2545
